examples/Server/Interceptors/UnaryCachingInterceptor.cs

```diff
@@ -0,0 +1,60 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using Grpc.Core;
+using Grpc.Core.Interceptors;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace Server.Interceptors
+{
+    public class UnaryCachingInterceptor : Interceptor
+    {
+        private static readonly MemoryCache _cache = new MemoryCache(new MemoryCacheOptions());
+        private readonly ILogger _logger;
+
+        public UnaryCachingInterceptor(ILogger<UnaryCachingInterceptor> logger)
+        {
+            _logger = logger;
+        }
+
+        public override async Task<TResponse> UnaryServerHandler<TRequest, TResponse>(TRequest request, ServerCallContext context, UnaryServerMethod<TRequest, TResponse> continuation)
+        {
+            if (_cache.TryGetValue<TResponse>(request, out var cachedResponse))
```

Comment:
This would require the the request type to implement Equals/GetHashCode, yes? e.g. HelloRequest is equal to another HelloRequest when the name is equal. Should probably have a comment explaining that

Reply:
Or do the generated types implement Equals/GetHashCode automatically?

Reply:
Yes the generated types implement this.